include/bluetooth/conn.h

```diff
@@ -277,20 +278,85 @@ int bt_conn_le_param_update(struct bt_conn *conn,
  */
 int bt_conn_disconnect(struct bt_conn *conn, u8_t reason);
 
+enum {
+	/** Convenience value when no options are specified. */
+	BT_LE_CONN_OPT_NONE = 0,
+
+	/* Enable scan on coded PHY, accept starting connection on CODED */
+	BT_LE_CONN_OPT_CODED = BIT(0),
+
+	/* Accept starting connection on 2M */
+	BT_LE_CONN_OPT_2M = BIT(1),
+
+	/* Disable scan on 1M PHY, disable starting connection on 1M. */
+	BT_LE_CONN_OPT_NO_1M = BIT(2),
+};
+
+struct bt_conn_le_create_param {
+
+	/** Bit-field of create connection options. */
+	u32_t options;
+
+	/** Scan interval (N * 0.625 ms) */
+	u16_t interval;
+
+	/** Scan window (N * 0.625 ms) */
+	u16_t window;
+
+	/** Scan interval LE Coded PHY (N * 0.625 MS)
+	 *  Set zero to use same as LE 1M PHY scan interval
+	 */
+	u16_t interval_coded;
+
+	/** Scan window LE Coded PHY (N * 0.625 MS)
+	 *  Set zero to use same as LE 1M PHY scan window.
+	 */
+	u16_t window_coded;
+};
+
+#define BT_CONN_LE_CREATE_PARAM(_options, _interval, _window) \
+	((struct bt_conn_le_create_param[]) { { \
+		.options = (_options), \
+		.interval = (_interval), \
+		.window = (_window), \
+		.interval_coded = 0, \
+		.window_coded = 0, \
+	 } })
+
+#define BT_CONN_LE_CREATE_CONN \
+	BT_CONN_LE_CREATE_PARAM(BT_LE_CONN_OPT_NONE, \
+				BT_GAP_SCAN_FAST_INTERVAL, \
+				BT_GAP_SCAN_FAST_INTERVAL)
+
+#define BT_CONN_LE_CREATE_CONN_AUTO \
+	BT_CONN_LE_CREATE_PARAM(BT_LE_CONN_OPT_NONE, \
+				BT_GAP_SCAN_FAST_INTERVAL, \
+				BT_GAP_SCAN_FAST_WINDOW)
+
 /** @brief Initiate an LE connection to a remote device.
  *
  *  Allows initiate new LE link to remote peer using its address.
  *  Returns a new reference that the the caller is responsible for managing.
  *
  *  This uses the General Connection Establishment procedure.
  *
- *  @param peer  Remote address.
- *  @param param Initial connection parameters.
+ *  @param peer         Remote address.
+ *  @param conn_param   Initial connection parameters.
+ *  @param create_param Create connection parameters.
  *
  *  @return Valid connection object on success or NULL otherwise.
  */
-struct bt_conn *bt_conn_create_le(const bt_addr_le_t *peer,
-				  const struct bt_le_conn_param *param);
+struct bt_conn *bt_conn_le_create(
```

Comment:
Since we're anyway breaking API here, I wonder if we should consider addressing #3936? In practice that'd mean adding an `int *err` parameter (unless someone has a better proposal?).

Reply:
Note: I'm not saying we *should* fix it, but this is now an opportunity to do it so we should consider it.

Reply:
Given that I think we need more time on this question. I would suggest removing the create_param for now, and when AE is enabled then have initiator on 1M + Coded always, since long-range connection establishment appear to be the requested use case that is most asked about.

Reply:
I say add an `int *err` in `struct bt_conn_le_create_param`. It is, after all, an "optional parameter".

Reply:
or even just an `int err` actually. No need for a pointer since we are already passing the pointer to the struct. That way we also ensure that the int exists and one less NULL pointer check

Reply:
@jhedberg and @joerchan please advise on the proposal above

Reply:
As agreed on slack discussion this has been made to have `int` as return value, and `struct bt_conn **conn` as out parameter.